src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int currentCapacity;
+    private int threshold;
+    private int size;
+    private Node<K,V>[] table;
+
+    private static class Node<K,V> {
+        private final K key;
+        private V value;
+        private Node<K,V> next;
+
+        Node(K key, V value, Node<K,V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    @SuppressWarnings("unchecked")
+    public MyHashMap() {
+        currentCapacity = DEFAULT_INITIAL_CAPACITY;
+        table = (Node<K, V>[]) new Node[currentCapacity];
+        threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int keyHash = hash(key);
+        Node<K, V> node = new Node<>(key, value, null);
+        if (table[keyHash] == null) {
+            table[keyHash] = node;
+            size++;
+        } else {
+            Node<K, V> existingNode = table[keyHash];
+            while (true) {
```

Comment:
```suggestion
           while (existingNode != null) {
```
You could drop if() on line 36, if you apply suggested changes